### DISCUSSION TITLE
reCAPTCHA

### DISCUSSION THREAD

**Discussion by Bit0r** [supabase/supabase#30747] [Feature Requests]
Does supabase support [google reCAPTCHA](https://developers.google.com/recaptcha/)? It is the most common CAPTCHA system in the world, with more users than `Hcaptcha` and `Cloudflare`, and the images of `reCAPTCHA` are easier to recognize than those of `Hcaptcha`, providing a better user experience.

**Comment by Bit0r** [comment]
Moreover, [reCAPTCHA v3](https://www.youtube.com/watch?v=tbvxFW4UJdU) does not require clicking on images; Google automatically detects whether a user is a robot through user behavior, greatly enhancing the user experience.

**Reply by Elsass1** [reply]
This would be indeed nice to have.

**Comment by anthonyisensee** [comment]
+1 this would be nice to have, especially with recaptcha v3 which does not requiring interaction. I will note that if v3 was implemented it would be nice to have a v2 fallback in case v3 fails. This gives the user a chance to manually verify they're not a bot.

**Comment by DanielSmith1239** [comment]
Has anyone tried using it? Or does Supabase need to explicitly add support?

**Comment by nasserahmed** [comment]
Any updates on this?

**Comment by robtaylordavis** [comment]
+1 for any updates please
